contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol

```diff
@@ -0,0 +1,289 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import {RouterBase, ITypeAndVersion} from "./RouterBase.sol";
+import {IFunctionsRouter} from "./interfaces/IFunctionsRouter.sol";
+import {IFunctionsCoordinator} from "./interfaces/IFunctionsCoordinator.sol";
+import {AuthorizedOriginReceiver} from "./accessControl/AuthorizedOriginReceiver.sol";
+import {IFunctionsSubscriptions, FunctionsSubscriptions, IFunctionsBilling} from "./FunctionsSubscriptions.sol";
+
+contract FunctionsRouter is RouterBase, IFunctionsRouter, AuthorizedOriginReceiver, FunctionsSubscriptions {
+  event RequestStart(bytes32 indexed requestId, Request request);
```

Comment:
It looks like this event is useless for o11y and FE, because of missing owner/sender/subid information we provide in `OracleRequest` event. I would wish we had just one "start" event and one "end" event that carry all the information at once.

Reply:
Okay, we may have two identical events in Router and in Coordinator, but if this event becomes the source for o11y, alerting and FE, we ultimately need all the same data we provide in OracleRequest event.

Reply:
Also, because this contract is supposed to be immutable, we cannot mention Request structure here, otherwise we won't be able to update Request structure.

Reply:
> Also, because this contract is supposed to be immutable, we cannot mention Request structure here, otherwise we won't be able to update Request structure.

The request structure is that which is used by the Router. The Coordinator has its own request structure. The Router one won't change.

Reply:
Let's talk about this tomorrow. `RequestStart` vs `OracleRequest` serve different purposes. I'm unsure of which o11y and FE should rely on.

Reply:
Is it possible that `OracleRequest` is fired, while `RequestStart` is not? If yes, then let's talk about this.
I understand `OracleRequest` is needed for the node, I am not suggesting removing it, but for o11y I would prefer to catch RequestStart/End events, because it does not need additional lookup for Coordinator address.

Reply:
I would rather separate the "internal state" `Request` struct and the event data. As a user I would expect to see my original parameters passed to request() function. This would also be great for o11y and we won't need to track `OracleRequest` event.